web/client/epics/geostory.js

```diff
@@ -164,21 +167,37 @@ export const scrollToContentEpic = action$ =>
 export const editMediaForBackgroundEpic = (action$, store) =>
     action$.ofType(EDIT_MEDIA)
         .switchMap(({path, owner}) => {
-            const state = store.getState();
-            const resourceId = resourceIdSelectorCreator(path)(state);
+            const selectedResource = resourceIdSelectorCreator(path)(store.getState());
             return Observable.of(
                 showMediaEditor(owner),
-                selectItem(resourceId)
+                selectItem(selectedResource)
             )
                 .merge(
                     action$.ofType(CHOOSE_MEDIA)
                         .switchMap( ({resource = {}}) => {
-                            const type = mediaTypeSelector(state);
-                            return Observable.of(
-                                update(`${path}`, {resourceId: resource.id, type}, "merge" )
-                            );
+                            let actions = [];
+                            const state = store.getState();
+                            const mediaType = currentMediaTypeSelector(state);
+                            const sourceId = sourceIdSelector(state);
+
+                            // if resources comes from geostory the id is at root level
+                            // otherwise it checks for the original resource id present in data
+                            const resourceAlreadyPresent = head(resourcesSelector(state).filter(r => r.data && (sourceId !== SourceTypes.GEOSTORY && r.data.id || r.id) === resource.id));
+
+                            let resourceId = resource.id;
+                            if (!resourceAlreadyPresent && resource.type === MediaTypes.MAP && sourceId !== SourceTypes.GEOSTORY) {
```

Comment:
Don't need to be a map. Every content from local source should not be added twice
```suggestion
                            if (!resourceAlreadyPresent && sourceId !== SourceTypes.GEOSTORY) {
```